com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v1/SpecialResourceTypesRequiredParameters.java

```diff
@@ -0,0 +1,34 @@
+package com.sap.cloud.lm.sl.cf.core.cf.v1;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+
+public enum SpecialResourceTypesRequiredParameters {
+
+    MANAGED_SERVICE_REQUIRED_PARAMETERS(ResourceType.MANAGED_SERVICE, SupportedParameters.SERVICE,
+        SupportedParameters.SERVICE_PLAN), USER_PROVIDED_SERVICE_REQUIRED_PARAMETERS(ResourceType.USER_PROVIDED_SERVICE,
+            SupportedParameters.SERVICE_CONFIG), EXISTING_SERVICE_REQUIRED_PARAMETERS(
+                ResourceType.EXISTING_SERVICE), EXISTING_SERVICE_KEY_REQUIRED_PARAMETERS(ResourceType.EXISTING_SERVICE_KEY);
+
+    private final ResourceType resourceType;
+    private final HashSet<String> requiredParameters = new HashSet<>();
+
+    SpecialResourceTypesRequiredParameters(ResourceType resourceType, String... requiredParameter) {
```

Comment:
Why package-private? Enum's should have private constructors. See https://stackoverflow.com/questions/7113363/private-enum-constructor.

Also, rename "requiredParameter" to "requiredParameters", since the variable technically can contain multiple parameters.